rare/components/tabs/games/__init__.py

```diff
@@ -159,146 +105,90 @@ def __init__(self, parent=None):
+                    rdlc.signals.progress.start.connect(rgame.signals.progress.start)
+                    rdlc.signals.progress.update.connect(rgame.signals.progress.update)
+                    rdlc.signals.progress.finish.connect(rgame.signals.progress.finish)
+                    rdlc.set_pixmap()
+                    rgame.owned_dlcs.append(rdlc)
+        return rgame
 
-        for game in self.no_assets:
-            if not game.metadata.get("customAttributes", {}).get("ThirdPartyManagedApp", {}).get("value") == "Origin":
-                icon_widget, list_widget = self.add_uninstalled_widget(game)
-            else:
-                icon_widget, list_widget = self.add_installed_widget(game.app_name)
+    def setup_game_list(self):
+        for game in self.api_results.games + self.api_results.na_games:
+            rgame = self.__create_game_with_dlcs(game)
+            self.rcore.add_game(rgame)
+            icon_widget, list_widget = self.add_library_widget(rgame)
             if not icon_widget or not list_widget:
-                logger.warning(f"Ignoring {game.app_name} in game list")
+                logger.warning(f"Excluding {rgame.app_name} from the game list")
                 continue
             self.icon_view.layout().addWidget(icon_widget)
             self.list_view.layout().addWidget(list_widget)
-
-        for game in sorted(self.game_list, key=lambda x: x.app_title):
-            if not self.core.is_installed(game.app_name):
-                self.uninstalled_games.append(game)
-                icon_widget, list_widget = self.add_uninstalled_widget(game)
-                self.icon_view.layout().addWidget(icon_widget)
-                self.list_view.layout().addWidget(list_widget)
+            rgame.set_pixmap()
         self.filter_games(self.active_filter)
+        self.update_count_games_label()
 
-    def add_installed_widget(self, app_name):
-        pixmap = self.image_manager.get_pixmap(app_name)
+    def add_library_widget(self, rgame: RareGame):
         try:
-            if pixmap.isNull():
-                logger.info(f"{app_name} has a corrupt image.")
-                if app_name in self.no_asset_names and self.core.get_asset(app_name).namespace != "ue":
-                    self.image_manager.download_image_blocking(self.core.get_game(app_name), force=True)
-                    pixmap = self.image_manager.get_pixmap(app_name)
-                elif self.ue_name:
-                    pixmap = self.image_manager.get_pixmap(self.ue_name)
-
-            icon_widget = InstalledIconWidget(app_name, pixmap, self.game_utils)
-            list_widget = InstalledListWidget(app_name, pixmap, self.game_utils)
+            icon_widget, list_widget = self.library_controller.add_game(rgame)
         except Exception as e:
-            logger.error(f"{app_name} is broken. Don't add it to game list: {e}")
+            raise e
+            logger.error(f"{rgame.app_name} is broken. Don't add it to game list: {e}")
```

Comment:
It does, previously we would not add certain games that error'd, currently they are handled more gracefully. The control flow it taken away because previously exceptions in those widgets would be hidden and it made it hard to debug. Good you mentioned it though since I want to remove that part before merging.